src/components/email-input.js

```diff
@@ -1,66 +1,64 @@
 import React from 'react';
-import autoBind from 'auto-bind';
 import PropTypes from 'prop-types';
 import { Form, Input } from 'antd';
-import { withUser } from './user-context.js';
-import { withTranslation } from 'react-i18next';
-import { userProps, translationProps, formItemLayoutShape } from '../ui/default-prop-types.js';
+import { useUser } from './user-context.js';
+import { useTranslation } from 'react-i18next';
+import { formItemLayoutShape } from '../ui/default-prop-types.js';
 
 const FormItem = Form.Item;
 
-class EmailInput extends React.Component {
-  constructor(props) {
-    super(props);
-    autoBind(this);
-  }
+function EmailInput({ forbiddenEmails, formItemLayout }) {
+  const user = useUser();
+  const { t } = useTranslation('emailInput');
 
-  normalize(email) {
+  const normalize = email => {
     return email ? email.toLowerCase() : email;
-  }
+  };
 
-  handleOnChange(event) {
+  const handleOnChange = event => {
     const element = event.target;
     const caret = element.selectionStart;
     window.requestAnimationFrame(() => {
       element.selectionStart = caret;
       element.selectionEnd = caret;
     });
-  }
+  };
 
-  render() {
-    const { forbiddenEmails, formItemLayout, user, t } = this.props;
-
-    const validationRules = [
-      {
-        required: true,
-        message: t('enterEmail')
-      },
-      {
-        type: 'email',
-        message: t('emailIsInvalid')
-      },
-      {
-        validator: (rule, value) => {
-          return value && forbiddenEmails.includes(value.toLowerCase())
-            ? Promise.reject(new Error(t('emailIsInUse')))
-            : Promise.resolve();
-        }
+  const validationRules = [
+    {
+      required: true,
+      message: t('enterEmail')
+    },
+    {
+      type: 'email',
+      message: t('emailIsInvalid')
+    },
+    {
+      validator: (rule, value) => {
+        return value && forbiddenEmails.includes(value.toLowerCase())
```

Comment:
Not related to the ticket, but it would be nice to have here `normalize(value)`, so it is consistent with the form item's normalization function (although it's pretty straight-forward)

Reply:
Sure, makes sense, change to use the function 👍